src/main/java/com/google/devtools/build/lib/analysis/skylark/SkylarkActionFactory.java

```diff
@@ -221,28 +221,58 @@ public void doNothing(String mnemonic, Object inputs) throws EvalException {
       SpawnInfo.spawnInfo;
 
   @Override
-  public void symlink(FileApi output, String path) throws EvalException {
+  public void symlink(FileApi output, Object target) throws EvalException {
     context.checkMutable("actions.symlink");
 
+    Artifact outputArtifact = (Artifact)output;
+    // TODO(yannic): This should be configurable.
+    String progressMessage = "Creating symlink " + outputArtifact.getRootRelativePathString();
+
+    if (target instanceof String) {
```

Comment:
@laurentlb , is this kind of behavior change based on the runtime type of the argument okay?

It seems to be reasonable on the surface, but there doesn't seem to be that many instances of this in our code and maybe explicitly specifying which one we want like `actions.symlink(output, file=<file>)` / `actions.symlink(output, symlink=<string>)` would be better?

Reply:
TBH, I actually like having `actions.symlink(output, file=<file>)` / `actions.symlink(output, symlink=<string>)` more than overloading `target` based on the runtime type.

Reply:
Same here, but I want this to be idiomatic, that's why I asked for @laurentlb 's opinion!